### ISSUE TITLE
Can we close all feature requests in joyent/node?

### ISSUE THREAD

**Issue by sam-github** [joyent/node#25544]
I'd like to close all 190 (!!!) feature requests: https://github.com/joyent/node/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request

0.12 should never land features that have not first landed in master (currently nodejs/io.js, but the repos are converging). Even then, I'd think some careful consideration by the TSC would be required before deciding to add a feature to the LTS branches (0.12 and 0.10, at the moment).

We could cut the issue count here by almost 25% by running through all feature requests and pasting in a nice message saying something to the effect of "If you still believe this to be relevant, re-open this request on nodejs/io.js. Thank you.", or something to that effect. I'd be willing to do this.

What do you all think, @joyent/node-collaborators ?

**Comment by cjihrig** [comment]
I'd be fine with closing them. It might be worth holding off until the converged repo is ready though, just so no one has to explain what's going on with the reconciliation to anyone not following along at home.

**Comment by jasnell** [comment]
Before considering closing, we should organize into three categories: 1)
won't ever do, 2) might do, 3) will do. The will do ones need corresponding
issues opened in nodejs/node to track the status. The might do ones may
need corresponding issues opened and the won't do ones simply get closed.

**Comment by sam-github** [comment]
@cjihrig 

That could be months away... and in the meantime feature requests pile up, and its hard to see the real bugs for the noise around them. Also, nodejs/io.js.git will be moved into the new location, preserving all issues/PRs, won't it? The explanatory text pasted could include a blurb about why this is necessary.

@jasnell 

We don't have to repost feature requests. There are 190! This is open source, the work is distributed, if the original poster wants to follow up, they can. I strongly suspect most of them no longer care, and there will not be hundreds of new feature requests in nodejs/io.js.

If nodejs wants to create a feature-request triaging system, it should be discussed there, based on the feature requests people are interested enough in to report there. That triaging doesn't have to happen in this repo.

I suggest this repo focus on LTS, anything else in here is a distraction. Also, there are more eyes on the nodejs/io.js repo, its better suited to feature request triage and commenting.

**Comment by jasnell** [comment]
I was planning to do a categorization of those over the next couple of
weeks. It shouldn't take long actually. Several of the feature request
tagged entries represent works in progress or things that are simply good
ideas that hadn't be gotten around to yet. We should not summarily close
those. I'm all for closing the ones we'll never actually do, but closing
them all without being sure is not a good approach IMHO.

**Comment by sam-github** [comment]
Well, we're just going to have to repeat this conversation once the feature requests hit nodejs/io.js or convergence, but here's my opinion, fwiw.

There are ***190***. There is, literally, no end to feature requests, not even an end to ones that would actually be accepted if well implemented. Leaving feature requests open is just an invitation for more. Closing a feature request doesn't delete it, nor does it erase its history, or make it impossible to comment on, or block its implementation in any way. If a feature has been comitted to by the TSC, or either the requestor or someone else clearly states an intention to implement the feature and made the request to gauge whether it would be acceptable, keeping it open makes sense. And of course, keep it open if it has a PR against it.

For anything else, IMO, it should be closed after a brief discussion period, with a comment giving an indication of whether it would be worth PRing such a feature. This should not be construed as unfriendly, just a "Great idea, reopen when you have some code, we're trying to keep the open issues focussed on bugs and work-in-progress, thanks!".

Btw, feedback from someone who has triaged a project with about the same number of issues:

http://words.steveklabnik.com/how-to-be-an-open-source-gardener

Note in particular:

> I went through a little algorithm to sort them out. It looked a little like this:
>
> 1. Is this issue a feature request? If so, copy/paste an answer I wrote that pointed them to the mailing list, and click close.

**Comment by sam-github** [comment]
And I should say, node uses github issue conversations, not mailing list, so I wouldn't literally point them to a mailing list (the rails project has a different structure than node in that respoect). The feature request issue itself is a good place for conversation, but that conversation doesn't require the issue to be open.

**Comment by jasnell** [comment]
I took some time yesterday to go through most of the feature requests. I closed quite a few and tagged others as "defer-to-converged". The next step is to take another pass and pull out all of the ones that deal primarily with documentation updates. I'll go through the documentation and see if the suggested changes make sense and get a single PR for landing those. Once that is submitted, I'll close all of the doc ones. That should cut it down quite a bit more. Ultimately I'll get the list shortened down to a list of items that we'll simply need to decide if we'll ever do at all or that need to be pushed over to the convergence repo.
